images/build-e2e/entrypoint.sh

```diff
@@ -128,7 +128,11 @@ if [ -n "${USER_PASSWORD}" ]; then
     OPTIONS+="--user-password=${USER_PASSWORD} "
 fi
 OPTIONS+="--godog.tags=\"@${PLATFORM} && ${E2E_TAG_EXPRESSION}\" --godog.format=junit "
-BINARY_EXEC+="cd ${EXECUTION_FOLDER}/bin && ./${BINARY} ${OPTIONS} > ${RESULTS_FILE}.results"
+if [[ ${PLATFORM} == 'windows' ]]; then
+    BINARY_EXEC+="cd ${EXECUTION_FOLDER}/bin && ./${BINARY} ${OPTIONS} > ${RESULTS_FILE}.results"
+else
+    BINARY_EXEC+="sudo su - ${TARGET_HOST_USERNAME} -c \"cd ${EXECUTION_FOLDER}/bin && ./${BINARY} ${OPTIONS} > ${RESULTS_FILE}.results\""
+fi
```

Comment:
When we are ssh to the machine/host are we ssh-ing as different user and that's the reason we need to use `sudo su - <username>`?

Reply:
Also in commit log `changin the user session` => `changing the user's session`

Reply:
No we are ssh-ing as same user, but doing the `sudo su -` we are changing the envs of the session from a remote ssh type of session to a local session (which is more aligned with a typical user case).

Reply:
Please mention that in the commit log, it now explains why we are doing it 👍🏼

Reply:
done